### PR TITLE
[TABLE MODEL] Implement CREATE/SHOW/DROP Function and user-defined scalar function

### PR DESCRIPTION
**Pull request by Cpaulyz**
## Description

https://timechor.feishu.cn/docx/GXnNdt3XIok3CQxBvwWcZxrnntd

This pull request introduces several new user-defined functions (UDFs) and corresponding integration tests for the IoTDB project. The changes include the addition of new scalar functions, updates to existing classes, and the creation of new test cases to ensure the functionality and robustness of the UDF management system.

### New Scalar Functions:
* [`ScalarFunctionExample`](diffhunk://#diff-293d1d808ec26cc6cd1b5b97fad76ca3b3d94d0199474914d345862a9065d285R1-R68): Implements a scalar function that checks if any input value is null and returns a boolean result.

### Integration Tests:
* [`IoTDBSQLFunctionManagementIT`](diffhunk://#diff-6f4560737c18ab9e2353dbac5105531946ee6f9b934352dfc6846182b4130e6aR1-R292): Adds multiple test cases to verify the creation, display, and deletion of scalar functions, including handling of invalid function names and URIs.

### CODE REVIEW

**Comment by lancelly** [comment]
What a spectacular work 🦾! Fabulous design and implementation, LGTM! ⚡️⚡️🔥

**Review by JackieTien97 — CHANGES_REQUESTED** [review]

File: iotdb-api/udf-api/src/main/java/org/apache/iotdb/udf/api/customizer/config/ScalarFunctionConfig.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * under the License.
+ */
+
+package org.apache.iotdb.udf.api.customizer.config;
+
+import org.apache.iotdb.udf.api.type.Type;
+
+public class ScalarFunctionConfig extends UDFConfigurations {
+
+  public ScalarFunctionConfig setOutputDataType(Type outputDataType) {
```
[the remark this answers is not in the document]
Comment (Cpaulyz) [reply iotdb-api/udf-api/src/main/java/org/apache/iotdb/udf/api/customizer/config/ScalarFunctionConfig.java:26]: Done

File: iotdb-api/udf-api/src/main/java/org/apache/iotdb/udf/api/customizer/parameter/FunctionParameters.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    return childExpressionDataTypes.get(index);
+  }
+
+  public boolean hasSystemAttribute(String attributeKey) {
+    return systemAttributes.containsKey(attributeKey);
+  }
+
+  public Map<String, String> getSystemAttributes() {
+    return systemAttributes;
+  }
```
[the remark this answers is not in the document]
Comment (Cpaulyz) [reply iotdb-api/udf-api/src/main/java/org/apache/iotdb/udf/api/customizer/parameter/FunctionParameters.java:55]: Done

File: iotdb-api/udf-api/src/main/java/org/apache/iotdb/udf/api/relational/ScalarFunction.java [hunk 3]
```diff
@@ -19,4 +19,50 @@
...
+  /**
+   * This method is used to validate {@link FunctionParameters}.
+   *
+   * @param parameters parameters used to validate
+   * @throws Exception if any parameter is not valid
+   */
+  void validate(FunctionParameters parameters) throws Exception;
+
+  /**
+   * This method is mainly used to initialize {@link ScalarFunction}. In this method, the user can
```
[the remark this answers is not in the document]
Comment (Cpaulyz) [reply iotdb-api/udf-api/src/main/java/org/apache/iotdb/udf/api/relational/ScalarFunction.java:37]: Done

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/execution/relational/ColumnTransformerBuilder.java [hunk 4]
```diff
@@ -996,6 +1003,25 @@ private ColumnTransformer getFunctionColumnTransformer(
...
+                Collections.emptyMap());
+        ScalarFunctionConfig config = new ScalarFunctionConfig();
+        scalarFunction.beforeStart(parameters, config);
+        Type returnType =
+            UDFDataTypeTransformer.transformUDFDataTypeToReadType(config.getOutputDataType());
+        return new UserDefineScalarFunctionTransformer(
+            returnType, scalarFunction, childrenColumnTransformer);
+      }
     }
     throw new IllegalArgumentException(String.format("Unknown function: %s", functionName));
```
[the remark this answers is not in the document]
Comment (Cpaulyz) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/execution/relational/ColumnTransformerBuilder.java:1026]: Done

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/execution/config/metadata/ShowFunctionsTask.java [hunk 5]
```diff
@@ -175,6 +260,15 @@ private static Binary getFunctionType(UDFInformation udfInformation) {
...
         }
+      } else {
+        if (TableUDFUtils.tryGetScalarFunction(udfInformation.getFunctionName()) != null) {
+          return BINARY_MAP.get(FUNCTION_TYPE_USER_DEFINED_SCALAR_FUNC);
+        } else if (TableUDFUtils.tryGetAggregateFunction(udfInformation.getFunctionName())
+            != null) {
+          return BINARY_MAP.get(FUNCTION_TYPE_USER_DEFINED_AGG_FUNC);
+        } else if (TableUDFUtils.tryGetTableFunction(udfInformation.getFunctionName()) != null) {
+          return BINARY_MAP.get(FUNCTION_TYPE_USER_DEFINED_TABLE_FUNC);
+        }
```
[the remark this answers is not in the document]
Comment (Cpaulyz) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/execution/config/metadata/ShowFunctionsTask.java:271]: Done~ Store FunctionType in UDFType now.

File: iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/udf/builtin/relational/TableBuiltinScalarFunction.java [hunk 6]
```diff
@@ -70,4 +75,14 @@ public enum TableBuiltinScalarFunction {
   public String getFunctionName() {
     return functionName;
   }
+
+  private static final Set<String> NATIVE_FUNCTION_NAMES =
```
[the remark this answers is not in the document]
Comment (Cpaulyz) [reply iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/udf/builtin/relational/TableBuiltinScalarFunction.java:79]: Done

File: iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/udf/builtin/relational/TableBuiltinScalarFunction.java [hunk 7]
```diff
@@ -70,4 +75,14 @@ public enum TableBuiltinScalarFunction {
...
+      new HashSet<>(
+          Arrays.stream(TableBuiltinScalarFunction.values())
+              .map(TableBuiltinScalarFunction::getFunctionName)
+              .collect(Collectors.toList()));
+
+  public static Set<String> getNativeFunctionNames() {
```
[the remark this answers is not in the document]
Comment (Cpaulyz) [reply iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/udf/builtin/relational/TableBuiltinScalarFunction.java:85]: Done

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/relational/metadata/TableMetadataImpl.java [hunk 8]
```diff
@@ -627,7 +630,26 @@ && isIntegerNumber(argumentTypes.get(2)))) {
         // ignore
     }
 
-    // TODO scalar UDF function
+    // User-defined scalar function
+    ScalarFunction scalarFunction = TableUDFUtils.tryGetScalarFunction(functionName);
```
[the remark this answers is not in the document]
Comment (Cpaulyz) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/relational/metadata/TableMetadataImpl.java:634]: I add a new interface `TableUDFUtils#isScalarFunction(String functionName)` to check syntax. 

`TableUDFUtils#getScalarFunction` will only be invoked if there is corresponding UDFInfomation on DataNode's UDFTable.  So we will not print warn log unless normally.

**Review by JackieTien97 — APPROVED** [review]

**Pull request merged**